### PR TITLE
Adding support for micro-benchmarks on CI

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

**Comment by sebastienros** [comment]
After that we can use the driver to run them:
`--projectFile benchmarks/Kestrel.Performance/Kestrel.Performance.csproj --benchmarkdotnet RequestParsingBenchmark --no-arguments --arg RequestParsingBenchmark --arg --config=perflab`

File: benchmarks/Kestrel.Performance/Kestrel.Performance.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <!-- Using shorter assembly name instead of Microsoft.AspNetCore.Server.Kestrel.Performance because https://github.com/dotnet/BenchmarkDotNet/issues/498 -->
     <AssemblyName>Kestrel.Performance</AssemblyName>
     <RootNamespace>Microsoft.AspNetCore.Server.Kestrel.Performance</RootNamespace>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (JunTaoLuo) [inline benchmarks/Kestrel.Performance/Kestrel.Performance.csproj]: Didn't we want to keep these at netcoreapp2.1 on the release/2.2 branch? https://github.com/aspnet/KestrelHttpServer/pull/3014#discussion_r225269003

**Comment by sebastienros** [comment]
I can make it more dynamic by adding conditions if necessary.

File: build/dependencies.props [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <InternalAspNetCoreSdkPackageVersion>2.2.0-preview2-20181011.2</InternalAspNetCoreSdkPackageVersion>
     <LibuvPackageVersion>1.10.0</LibuvPackageVersion>
     <MicrosoftAspNetCoreAllPackageVersion>2.2.0-preview3-35457</MicrosoftAspNetCoreAllPackageVersion>
-    <MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>2.2.0-preview3-35457</MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>
```
Comment (halter73) [inline build/dependencies.props]: Why not just `run.ps1 upgrade deps`?

Comment (sebastienros) [reply build/dependencies.props]: Screw that thing that installs random dotnet versions on my box, and now I get to develop using a custom build and see warnings where I code on stuff ;)

File: benchmarks/Kestrel.Performance/Kestrel.Performance.csproj (hunk 1 above)
Comment (JunTaoLuo) [reply benchmarks/Kestrel.Performance/Kestrel.Performance.csproj]: I misinterpreted that comment.

**Pull request merged**